src/documents/markdown.ts

```diff
@@ -1,7 +1,12 @@
 import { writeFileSync } from 'fs';
 import { join } from 'path';
 import { heading } from '../elements/heading.js';
-import { ContentSection } from '../sections/content-section.js';
+import {
+  getSection,
+  Section,
+  type InferSection,
```

Comment:
I do not import types this way - is the main benefit clarity? I will use type ocasionally when I am only importing a type.

Reply:
At this point it's force of habit, I do it all the time now. There have been a couple of tooling related issues that aren't present when it's explicitly a type import. Circular dependencies is one issue, not that that would be the case here. Another is complaints about dev deps having to be runtime dependencies just to use the types.